corpus/transform.py

```diff
@@ -491,3 +491,73 @@ def run(self):
 
         with open(str(self.out_segments), "w") as segments_outfile:
             segments_outfile.writelines(segment_file_names)
+
+
+class LexiconStrategy(enum.Enum):
+    PICK_FIRST = 0
+
+
+class ApplyLexiconToTranscriptionsJob(Job):
+    """
+    Use a bliss lexicon to convert all words in a bliss corpus into their phoneme representation.
+
+    Currently only supports picking the first phoneme.
+    """
+
+    def __init__(
+        self,
+        bliss_corpus,
+        bliss_lexicon,
+        word_separation_orth,
+        strategy=LexiconStrategy.PICK_FIRST,
+    ):
+        """
+        :param bliss_corpus: path to a bliss corpus xml
+        :param bliss_lexicon: path to a bliss lexicon file
+        :param str|None word_separation_orth: a default word separation lemma orth. The corresponding phoneme
+            (or phonemes in some special cases) are inserted between each word.
+            Usually it makes sense to use something like "[SILENCE]" or "[space]" or so).
+        :param LexiconStrategy strategy: strategy to determine which representation is selected
+        """
+        self.bliss_corpus = bliss_corpus
+        self.bliss_lexicon = bliss_lexicon
+        self.word_separation_orth = word_separation_orth
+        self.strategy = strategy
+
+        self.out_corpus = self.output_path("corpus.xml.gz")
+
+    def tasks(self):
+        yield Task("run", mini_task=True)
+
+    def run(self):
+        c = corpus.Corpus()
+        c.load(self.bliss_corpus.get_path())
+
+        lex = lexicon.Lexicon()
+        lex.load(self.bliss_lexicon.get_path())
+
+        # build lookup dict
+        lookup_dict = {}
+        for lemma in lex.lemmata:
+            for orth in lemma.orth:
+                if orth and self.strategy == LexiconStrategy.PICK_FIRST:
+                    if len(lemma.phon) > 0:
+                        lookup_dict[orth] = lemma.phon[0]
+
+        if self.word_separation_orth is not None:
+            word_separation_phon = lookup_dict[self.word_separation_orth]
+            print("using word separation symbold: %s" % word_separation_phon)
+            separator = " %s " % word_separation_phon
```

Comment:
just to be sure: is the space behind the `word_separation_phon` intended?

Reply:
Yes, this is intended for `separator.join(words)`.